### PR TITLE
[T267712.2] Add job that outputs versions used in build

### PR DESCRIPTION
**Pull request by toban**
- Adds PHP_VERSION env var
- Adds workflow step versions that writes built_versions.log to metadata with the versions to be used in https://phabricator.wikimedia.org/T267712

### CODE REVIEW

File: build/requirements/build_version_requirements.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+def getVar( varName ):
+    if varName in os.environ:
+        name = components[varName]
+        return name + " " + os.environ[varName]
+    else:
+        return 'undefined'
+
+components =	{
+  "RELEASE_MAJOR_VERSION": "Mediawiki",
```
Comment (mariushoch) [inline build/requirements/build_version_requirements.py:12]: MediaWiki

File: build/requirements/build_version_requirements.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+with open('config.yml') as f:
+  data = yaml.load(f, Loader=yaml.FullLoader)
+  databases = data['jobs']['test_wikibase']['strategy']['matrix']['databaseImageName']
+
+f = open("artifacts/built_versions.log", "w+")
+
+for key in components:
+  f.write(getVar(key) + '\n')
+
+f.write('Tested databases: ' + ', '.join(databases))
```
Comment (mariushoch) [inline build/requirements/build_version_requirements.py:30]: Not that it matters much, but the other lines are just `component version`, but this one is `component: versions`… do we maybe want to use `component: version` everywhere?

Comment (toban) [reply build/requirements/build_version_requirements.py]: sounds good

File: build/requirements/build_version_requirements.py (hunk 1 above)
Comment (toban) [reply build/requirements/build_version_requirements.py]: done

**Review by mariushoch — APPROVED** [review]

**Pull request merged**